File: src/main/java/com/divide2/dic/model/Dic.java
package com.divide2.dic.model;

import com.divide2.search.annotation.Conditioner;
import com.divide2.search.annotation.SearchWay;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;

import javax.persistence.*;

/**
 * Created by bvvy on 2018/1/4.
 * com.divide2.dic.model
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "sys_dic")
@Document(indexName = "dics",type = "dic",refreshInterval = "-1")
public class Dic {

    @Id
    @GeneratedValue
    @org.springframework.data.annotation.Id
    @Conditioner(way = SearchWay.RANGE,startName = "idStart",endName = "idEnd")
    private Integer id;

    @Column(name = "group_name")
    private String groupName;


    @Column(name = "group_code")
    @Conditioner(way = SearchWay.LIKE)
    private String groupCode;

    @Column(name = "cn_label")
    private String cnLabel;

    @Column(name = "en_label")
    private String enLabel;

    @Column(name = "order_num")
    private Integer orderNum;

    private String code;
}

File: src/main/kotlin/com/divide2/dic/Dic.kt
package com.divide2.dic


import javax.persistence.*

/**
 * Created by bvvy on 2018/1/4.
 * com.divide2.dic.model
 */
@Entity
@Table(name = "sys_dic")
data class Dic(
        @Id
        @GeneratedValue
        var id: Int = 0,

        @Column(name = "group_name")
        var groupName: String = "",

        @Column(name = "group_code")
        var groupCode: String = "",

        @Column(name = "cn_label")
        var cnLabel: String = "",

        @Column(name = "en_label")
        var enLabel: String = "",

        @Column(name = "order_num")
        var orderNum: Int = 0,

        var code: String = ""
)

